compiler/ast/src/ast_gen.rs

```diff
@@ -195,7 +195,7 @@ pub enum ExprKind<U = ()> {
         orelse: Box<Expr<U>>,
     },
     Dict {
-        keys: Vec<Expr<U>>,
+        keys: Vec<Option<Expr<U>>>,
```

Comment:
I found I changed it to the previous form to keep cpython compatibility in #4042.
I'd like to ask `Option<Expr<U>>` is fit in cpython way.
I thought this is the new cpython way when i was working on #4042 which is changed in 3.10

Reply:
@youknowone Why did you change it to the previous form?

Reply:
I can't remember well so looking in it again.
I thought CPython's Python.asdl actually changed it to not optional here.
https://github.com/RustPython/RustPython/pull/4042/files#diff-242a44b9d53b6c6212a752e0c97d6cf2f87794f579700aa8015bb99cef63925cR63

But I still can see ast.parse generate None for empty keys in Python 3.10. I am confused.
Do you know what the '?' means in `Python.asdl`?

Reply:
I think `?` means optional.

Reply:
Yes, that was the reason I thought CPython changed it from optional key to non-optional one in 3.10. How they still have None in parse.dump?